boost/teamcity_boost.cpp

```diff
@@ -17,21 +17,33 @@
 
 #include <sstream>
 
-#include <boost/test/unit_test_suite_impl.hpp>
 #include <boost/test/results_collector.hpp>
 #include <boost/test/utils/basic_cstring/io.hpp>
 #include <boost/test/unit_test_log.hpp>
+#include <boost/test/execution_monitor.hpp>
 #include <boost/test/unit_test_log_formatter.hpp>
 #include <boost/test/unit_test.hpp>
 
+// In 1.59.0, they changed the name of this enum value, so we have to this hacky thing...
+#include <boost/version.hpp>
+#if BOOST_VERSION >= 105900
+    #define TUT_CASE_IDENTIFIER boost::unit_test::TUT_CASE
+#else
+    #define TUT_CASE_IDENTIFIER boost::unit_test::tut_case
+#endif
+
 #include "teamcity_messages.h"
 
 namespace jetbrains { namespace teamcity {
+const std::string ASSERT_CTX = "Assertion occurred in a following context:";
+const std::string FAILURE_CTX = "Assertion occurred in a following context:";
 
 // Custom formatter for TeamCity messages
 class TeamcityBoostLogFormatter: public boost::unit_test::unit_test_log_formatter {
     TeamcityMessages messages;
     std::string currentDetails;
+    std::string currentContextDetails;
+    std::string currentTestName;
```

Comment:
you don't need this actually. it is always possible to get a current test name. see my fork of this...